src/query/storages/fuse/src/io/read/meta_readers.rs

```diff
@@ -96,17 +115,28 @@ where T: BufReaderProvider + Sync + Send
 impl<T> Loader<SegmentInfo> for LoaderWrapper<T>
 where T: BufReaderProvider + Sync + Send
 {
-    async fn load(&self, key: &str, length_hint: Option<u64>, version: u64) -> Result<SegmentInfo> {
+    async fn load(
+        &self,
+        key: &str,
+        length_hint: Option<u64>,
+        sp: Option<StorageParams>,
+        version: u64,
+    ) -> Result<SegmentInfo> {
         let version = SegmentInfoVersion::try_from(version)?;
-        let reader = self.0.buf_reader(key, length_hint).await?;
+        let reader = self.0.buf_reader(key, length_hint, sp).await?;
         version.read(reader).await
     }
 }
 
 #[async_trait::async_trait]
 impl BufReaderProvider for &dyn TableContext {
-    async fn buf_reader(&self, path: &str, len: Option<u64>) -> Result<BufReader<BytesReader>> {
-        let operator = self.get_storage_operator()?;
+    async fn buf_reader(
+        &self,
+        path: &str,
+        len: Option<u64>,
+        sp: Option<StorageParams>,
+    ) -> Result<BufReader<BytesReader>> {
+        let operator = self.get_storage_operator(sp)?;
```

Comment:
Cloning an operator is cheap, but building an operator is far more expensive, especially for services like s3.

Try our best to reuse the same operator.